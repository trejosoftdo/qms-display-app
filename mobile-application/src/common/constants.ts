export const AUTHORIZATION_PENDING_CODE = 'authorization_pending';

export const VALIDATION_ERROR_STATUS_CODE = 400;

export const DEVICE_NOT_CONNECTED_ERROR_MESSAGE = 'Device is not connected';

export const CONNECTION_EXPIRED_ERROR_MESSAGE = 'Connection expired!';

export const APP_PROTO = 'app://';

export const DEVICE_ID_KEY = 'SECURE_DEVICE_ID';

export const CONNECTION_DETAILS_KEY = 'CONNECTION_DETAILS';

export const WEB_PLATFORM = 'web';

export const BEARER_PORTION = 'Bearer';

export const EMPTY_VALUE = '';

export const DEFAULT_REGULAR_FONT_NAME = 'Montserrat-Regular';

export const DEFAULT_LIGHT_FONT_NAME = 'Montserrat-Light';

export const DEFAULT_MEDIUM_FONT_NAME = 'Montserrat-Medium';

export const NORMAL_FONT_WEIGHT = 'normal';

export const ONE_SECOND_MILISECONDS = 1000;

export const INTERVAL_TIME = 10 * ONE_SECOND_MILISECONDS;

export const SERVICES_PATH = '/services';

export const TICKET_DETAILS_PATH = '/ticket-details';

export const HOME_PATH = '/';

export const CATEGORIES_PATH = '/categories';

export const CONFIGURATION_PATH = '/configuration';

export const NO_OP = () => {};


export const LANGUAGES = {
  SPANISH: 'es',
  ENGLISH: 'en',
};

export const CONFIG_OPTIONS = {
  INFORMATION: 'information',
  CONNECTION: 'connection',
  LANGUAGE: 'language',
};
