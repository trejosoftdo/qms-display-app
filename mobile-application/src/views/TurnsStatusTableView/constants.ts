export const BEING_ATTENDED_STATUS_CODE = "BEING_ATTENDED";

export const TO_BE_ATTENDED_STATUS_CODE = "TO_BE_ATTENDED";