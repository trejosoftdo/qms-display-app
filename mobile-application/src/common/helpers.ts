import { Platform } from 'react-native';
import { router } from 'expo-router';
import { APP_PROTO, EMPTY_VALUE, ONE_SECOND_MILISECONDS, WEB_PLATFORM } from './constants';

/** 
 * Checks if the platform is web
 * @returns boolean
 */
export const isWeb = (): boolean => Platform.OS === WEB_PLATFORM;

/** 
 * Gets the current time
 * @returns number
 */
export const getCurrentTime = (): number => new Date().getTime();

/**
 * Calculates the expire time
 * 
 * @param  {number} expiresIn
 * @returns number
 */
export const calculateExpireTime = (expiresIn: number): number => getCurrentTime() + expiresIn * ONE_SECOND_MILISECONDS;


/** 
 * Extracs icon name
 * @returns string
 */
export const extractIconName = (iconUrl: string): string => iconUrl.replace(APP_PROTO, EMPTY_VALUE)

/**
 * Navigates to a given path
 * 
 * @param  {string} pathname
 * @param  {any} params
 * @returns void
 */
export const goToPath = (pathname: string, params: Record<string, any> = undefined): void => {
  router.push({
    pathname,
    params,
  });
};

/**
 * Waits for certain time
 * @param  {number} timeout 
 * @returns Promise<any>
 */
export const delay = (timeout: number) => new Promise((resolve) => {
  setTimeout(resolve, timeout);
});

/**
 * Gets the response blob url
 * @param  {Response} response 
 * @returns Promise<string>
 */
export const getResponseBlobUrl = async (response: Response): Promise<string> => {
  const stream = await response.blob();
  return URL.createObjectURL(stream);
};
