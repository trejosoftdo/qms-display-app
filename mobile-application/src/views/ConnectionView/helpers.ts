import { ConnectionData, ConnectionDetails } from "../../common/models";
import { TRANSLATION_CONNECTED_MESSAGE_KEY, TRANSLATION_INITIAL_CONNECT_MESSAGE_KEY, TRANSLATION_SUCCESS_CONNECTION_MESSAGE_KEY, TRANSLATION_TO_CONNECT_MESSAGE_KEY } from "../../common/translations/translation-keys";

/**
 * Gets the correct message key from the data state
 * 
 * @param  {ConnectionData} data
 * @returns string
 */
export const getMessageKey = (data: ConnectionData, connectionDetails?: ConnectionDetails): string => {
  if (connectionDetails?.accessToken) {
    return TRANSLATION_CONNECTED_MESSAGE_KEY;
  }

  if (data?.tokens?.accessToken) {
    return TRANSLATION_SUCCESS_CONNECTION_MESSAGE_KEY;
  }

  return data?.userCode ? TRANSLATION_TO_CONNECT_MESSAGE_KEY : TRANSLATION_INITIAL_CONNECT_MESSAGE_KEY;
};
