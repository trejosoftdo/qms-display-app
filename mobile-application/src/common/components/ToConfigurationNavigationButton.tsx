import { FC } from 'react';
import { StyleSheet } from 'react-native';
import { IconButton } from 'react-native-paper';
import { goToPath } from '../helpers';
import { CONFIGURATION_PATH } from '../constants';
import AppLightTheme from '../theme';

/**
 * ToConfigurationNavigationButtonProps defines the props for the To Configuration Navigation Button Component.
 */
interface ToConfigurationNavigationButtonProps {}

/**
 * A button that when clicked goes to configuration section
 *
 * @param {ToConfigurationNavigationButtonProps} props - The props for the To Configuration Navigation Button component.
 */
const ToConfigurationNavigationButton: FC<ToConfigurationNavigationButtonProps> = (props: ToConfigurationNavigationButtonProps) => (
  <IconButton
    icon="cog"
    iconColor={AppLightTheme.colors.background}
    size={20}
    onPress={() => {
      goToPath(CONFIGURATION_PATH);
    }}
  />
);


export default ToConfigurationNavigationButton;
