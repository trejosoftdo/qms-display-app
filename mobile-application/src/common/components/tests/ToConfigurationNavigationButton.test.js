import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import ToConfigurationNavigationButton from '../ToConfigurationNavigationButton';
import { goToPath } from '../../helpers';
import { CONFIGURATION_PATH } from '../../constants';


jest.mock('../../helpers');

describe('ToConfigurationNavigationButton component', () => {
  it('renders the component as expected', () => {
    expect(renderShallow(<ToConfigurationNavigationButton />)).toMatchSnapshot();
  });

  it('moves to configuration path when being clicked', () => {
    const { getByRole } = render(<ToConfigurationNavigationButton />);
    const element = getByRole('button');
    fireEvent.press(element);
    expect(goToPath).toHaveBeenCalledTimes(1);
    expect(goToPath).toHaveBeenCalledWith(CONFIGURATION_PATH);
  });
});
