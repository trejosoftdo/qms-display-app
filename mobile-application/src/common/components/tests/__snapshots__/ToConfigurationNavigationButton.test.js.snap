// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToConfigurationNavigationButton component renders the component as expected 1`] = `
<ForwardRef
  icon="cog"
  iconColor="#ffffff"
  onPress={[Function]}
  size={20}
/>
`;
